src/blocks/frontend/live-search/index.ts

```diff
@@ -0,0 +1,252 @@
+/**
+ * Wordpress dependencies
+ */
+
+// @ts-ignore
+import { debounce } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { domReady } from '../../helpers/frontend-helper-functions.js';
+
+type ResultsEntry = {
+	link: string,
+	title: string
+}
+
+type ResultsContainer = Element | null | undefined;
+
+domReady( () => {
+	const CONTAINER_CLASS = 'search-results';
+
+	// @ts-ignore
+	const { nonce, restUrl, strings } = liveSearchData;
+	const liveSearch = document.querySelectorAll( '.o-live-search' );
+	const loadingIcon = '<svg class="spinner" viewBox="0 0 100 100" width="16" height="16" xmlns="http://www.w3.org/2000/svg" focusable="false" style="width: calc(16px); height: calc(16px);"><circle cx="50" cy="50" r="50" vector-effect="non-scaling-stroke" class="main-circle"></circle><path d="m 50 0 a 50 50 0 0 1 50 50" vector-effect="non-scaling-stroke" class="moving-circle"></path></svg>';
+
+	const requestData = async( search: string, postTypes: Array<string> ) => {
+		const options = {
+			method: 'GET',
+			headers: {
+				Accept: 'application/json',
+				'Content-Type': 'application/json',
+				'x-wp-nonce': nonce
+			}
+		};
+
+		let params = `s=${search}`;
+		postTypes.forEach( type => {
+			params = `${params}&post_types[]=${type}`;
+		});
```

Comment:
When it comes to reducing a list of things to a single value, I prefer using [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) since it has a better-defined scope, thus it is easily understood on a quick look 👀 . 

```javascript
const params = postTypes.reduce((p, type) => p + `&post_types[]=${type}`, `s=${search}`)
```